lib/packwerk/application_validator.rb

```diff
@@ -170,14 +170,15 @@ def check_application_structure
     def check_inflection_file
       inflections_file = @configuration.inflections_file
 
-      test_inflections = ActiveSupport::Inflector::Inflections.new
+      application_inflections = ActiveSupport::Inflector::Inflections.new
+      Packwerk::Inflections::Default.apply_to(application_inflections)
+      Packwerk::Inflections::Custom.new(inflections_file).apply_to(application_inflections)
```

Comment:
I'm pretty sure this is wrong.

The validator is running in the application context, so `application_inflections = ActiveSupport::Inflector.inflections`.

Reply:
Thanks @exterm. Since Maple’s away this week I’m going to try to resolve this on her behalf.

Reply:
Okay, I agree — this is the crux of my [confusion](https://github.com/Shopify/packwerk/pull/53#discussion_r514148499) over this bit of code (“I don’t understand under what circumstances this check is intended to fail”).

I think we’re trying to confirm that the application has [correctly wired up](https://github.com/Shopify/packwerk/blob/main/USAGE.md#inflections) its Packwerk inflections in `config/initializers/inflections.rb` or wherever. But if we ourselves apply the Packwerk inflections on top of whatever the application has already loaded, this check will always succeed regardless of whether the application is wired up correctly.

As far as I can tell we have no unit or acceptance tests covering the case where `#check_inflection_file` fails, so it’s unsurprising that it doesn’t work. It’s a separate bug than the one being fixed by this PR, so I’ll open another PR and rebase this one on top of it.

Reply:
Oh, surprise, I was confused again. The code on `main` does the right thing, it’s just this change that’s wrong. Sorry. I shall fix this PR then.